File: ios/SpentyAI/SpentyAI/Core/Models/Bill.swift
import Foundation

struct Bill: Codable, Identifiable {
    let id: String
    var billNumber: String?
    var vendorId: String?
    var vendorName: String?
    var date: Date?
    var dueDate: Date?
    var lineItems: [BillLineItem]?
    var subtotal: Double?
    var taxAmount: Double?
    var grandTotal: Double?
    var paymentStatus: String?
    var amountPaid: Double?
    var payments: [BillPayment]?
    var notes: String?

    enum CodingKeys: String, CodingKey {
        case id = "billId"
        case billNumber
        case vendorId
        case vendorName
        case date
        case dueDate
        case lineItems
        case subtotal
        case taxAmount
        case grandTotal
        case paymentStatus
        case amountPaid
        case payments
        case notes
    }
}

struct BillLineItem: Codable, Identifiable {
    var id: String { description ?? UUID().uuidString }
    var description: String?
    var quantity: Double?
    var rate: Double?
    var amount: Double?
    var taxPercent: Double?

    enum CodingKeys: String, CodingKey {
        case description
        case quantity
        case rate
        case amount
        case taxPercent
    }
}

struct BillPayment: Codable, Identifiable {
    var id: String { date?.description ?? UUID().uuidString }
    var amount: Double?
    var date: Date?
    var method: String?
    var note: String?

    enum CodingKeys: String, CodingKey {
        case amount
        case date
        case method
        case note
    }
}

File: android-native/app/src/main/java/com/spentyai/app/core/models/Bill.kt
package com.spentyai.app.core.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Bill(
    @SerialName("bill_id") val id: String,
    @SerialName("bill_number") val billNumber: String? = null,
    @SerialName("vendor_id") val vendorId: String,
    @SerialName("vendor_name") val vendorName: String? = null,
    val status: BillStatus = BillStatus.UNPAID,
    @SerialName("issue_date") val issueDate: String,
    @SerialName("due_date") val dueDate: String,
    val subtotal: Double = 0.0,
    @SerialName("tax_amount") val taxAmount: Double = 0.0,
    val total: Double = 0.0,
    @SerialName("amount_paid") val amountPaid: Double = 0.0,
    @SerialName("amount_due") val amountDue: Double = 0.0,
    val currency: String = "USD",
    val notes: String? = null,
    @SerialName("line_items") val lineItems: List<BillLineItem>? = null,
    @SerialName("payment_date") val paymentDate: String? = null,
    @SerialName("category_id") val categoryId: String? = null,
    @SerialName("category_name") val categoryName: String? = null,
    @SerialName("is_recurring") val isRecurring: Boolean = false,
    @SerialName("recurrence_pattern") val recurrencePattern: String? = null,
    @SerialName("created_at") val createdAt: String? = null,
    @SerialName("updated_at") val updatedAt: String? = null
)

@Serializable
enum class BillStatus {
    @SerialName("unpaid") UNPAID,
    @SerialName("paid") PAID,
    @SerialName("overdue") OVERDUE,
    @SerialName("partially_paid") PARTIALLY_PAID,
    @SerialName("cancelled") CANCELLED
}

@Serializable
data class BillLineItem(
    val description: String,
    val quantity: Double = 1.0,
    @SerialName("unit_price") val unitPrice: Double = 0.0,
    val amount: Double = 0.0,
    @SerialName("tax_rate") val taxRate: Double? = null
)
